products/views.py

```diff
@@ -1,3 +1,43 @@
-from django.shortcuts import render
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse_lazy
+from django.views.generic import ListView, FormView
+from .forms import ProductForm
+from .models import Product
+from common.mixins import TitleMixin
+from django.contrib.auth.mixins import LoginRequiredMixin
 
-# Create your views here.
+
+class ProductListView(TitleMixin, ListView):
+    title = 'Tech Guru'
+    model = Product
+    template_name = 'product_list.html'
+    context_object_name = 'products'
+
+
+class AddProductView(LoginRequiredMixin, TitleMixin, FormView, ListView):
+    title = 'Add Product'
+    model = Product
+    form_class = ProductForm
+    template_name = 'add_product.html'
+    success_url = reverse_lazy('product_list')
+    success_message = "Product Added Successfully"
+    login_url = reverse_lazy('login')
+
+    def form_valid(self, form):
```

Comment:
You dont need rewrite this function, becouse inside `FormView` its already exists, with same logics